viper.go

```diff
@@ -280,76 +279,67 @@ func WatchConfig() { v.WatchConfig() }
 
 // WatchConfig starts watching a config file for changes.
 func (v *Viper) WatchConfig() {
-	initWG := sync.WaitGroup{}
-	initWG.Add(1)
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		v.logger.Error(fmt.Sprintf("failed to create watcher: %s", err))
+		panic(err)
```

Comment:
The log is "clothed", but the panic is "naked".

I would like yo suggest something like this.

This way the panic will be easier to understand


```suggestion
		err = fmt.Errorf("failed to create watcher: %w", err)
		v.logger.Error(err.Error())
		panic(err)
```

This suggestion should be applied everywhere

Reply:
I wasn't sure if I should wrap the error or not when panicking it. I always feel a bit weird immediately re-definig the error to wrap it after it was just set, but yeah, I think it makes sense to do here given the panic would need to be investigated from other code. Will get that fixed up.